doc/page-properties.h

```diff
@@ -0,0 +1,145 @@
+/**********************************************************************
+**
+** Copyright (C) Luxoft Sweden AB 2020
+**
+** This file is part of the documentation of the FaceLift project
+**
+** This work is licensed under the Creative Commons
+** Attribution-ShareAlike 4.0 International License. To view a copy of
+** this license, visit http://creativecommons.org/licenses/by-sa/4.0/
+** or send a letter to Creative Commons, PO Box 1866, Mountain View,
+** CA 94042, USA.
+**
+** SPDX-License-Identifier: CC-BY-SA-4.0
+**
+**********************************************************************/
+
+/**
+\page page-properties Properties
+
+\section properties-intro Introduction
+Interface definition files (*.qface) can define properties. See https://github.com/Pelagicore/qface for information
+about QFace interface definition.
+
+In Facelift each property is exposed to the QML engine as Q_PROPERTY with a setter and getter functions, and a change
+signal. If the property is marked as 'readonly' then no setter is generated.
+
+In Facelift properties are generated as members of the &lt;Inerface name&gt;ImplementationBase C++ class.
+The facelift::Property class template provides encapsulation for the internal data. To get the underlying value
+developer can use a getter that returns the const reference to the underlying data (in case of the service property
+it is const reference to a pointer) or one of the specialization specific access functions. The value can be changed only by
+using a setter or one of the specialization specific access functions. This guarantees that the data will not be unnoticibly
+ changed. Each time the setter is called, with a value that differs from the previous one, the change signal is emitted.
+
+\section QML
+
+In case of C++ interface implementation, the property, genetated as a C++ code, is exposed directly to the QML engine
+by Q_PROPERTY macro and can be used without any additional effort.
+
+\section cpp C++
+
+The property in C++ is generated as one of the templated property containers: Property<T>, ListProperty<T>,
+ServiceProperty<T> or ModelProperty<T>. Additionally ListProperty and QMap<QString, T> partial template specializations
+add extra convenience functions: void removeAt(int i), void addElement(ElementType element) and int size() const.
+The QMap<QString, T> specialization is created for the map property defined in *.qface file.
+
+\section subservice Property ownership
+
+The service owns the property objects and maintains their life cycle.
+
+The service property returns a pointer to another service. In this case property object is not an owner of the
```

Comment:
"The service property" ?? which one ? a property does not return anything

Reply:
to be changed